src/System/Contexts/GlobalMeContext.tsx

```diff
@@ -0,0 +1,87 @@
+import { Suspense, createContext, useEffect, useState } from "react"
+import { usePreloadedQuery, useQueryLoader } from "react-relay/hooks"
+import { graphql } from "react-relay"
+import {
+  GlobalMeContextQuery,
+  GlobalMeContextQuery$data,
+} from "__generated__/GlobalMeContextQuery.graphql"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+
+export interface GlobalMeContextProps {
+  me: NonNullable<GlobalMeContextQuery$data["me"]> | null
+  refetchMe: () => void
+}
+
+export const GlobalMeContext = createContext<GlobalMeContextProps>({
+  me: null,
+  refetchMe: () => {},
+})
+
+const GLOBAL_ME_QUERY = graphql`
+  query GlobalMeContextQuery {
+    me {
+      ...useCollectorSignals_me
+    }
+  }
+`
+
+const GlobalMeQuery = ({ queryRef, children, refetchMe }) => {
+  const data = usePreloadedQuery<GlobalMeContextQuery>(
+    GLOBAL_ME_QUERY,
+    queryRef
+  )
+
+  const me = data?.me ?? null
+
+  return (
+    <GlobalMeContext.Provider
+      value={{
+        me,
+        refetchMe,
+      }}
+    >
+      {children}
+    </GlobalMeContext.Provider>
+  )
+}
+
+export const GlobalMeProvider = ({ children }) => {
+  const { user, relayEnvironment } = useSystemContext()
+  const [loaded, setLoaded] = useState(false)
+  const [queryRef, loadQuery] = useQueryLoader(GLOBAL_ME_QUERY)
+
+  const refetchMe = () => loadQuery({})
+  console.log("*** GlobalMeProvider prefetch", {
+    user,
+    relayEnvironment,
+    loaded,
+    queryRef,
+    loadQuery,
+  })
+  useEffect(() => {
+    if (!loaded) {
+      if (!!user) {
+        console.log("*** Loading ME query")
+        loadQuery({})
+      } else {
+        console.log("skipping loading me")
+      }
+      setLoaded(true)
+    }
+  }, [loadQuery, loaded, user])
+
+  if (!!user && !queryRef) {
+    return "...Loading"
+  }
+  if (!user) {
+    return children
+  }
+
+  return (
+    <Suspense fallback="Loading ...">
+      <GlobalMeQuery queryRef={queryRef} refetchMe={refetchMe}>
+        {children}
+      </GlobalMeQuery>
+    </Suspense>
+  )
```

Comment:
My pitch, but very open to having this torn apart: A global `Me` context.
- First time with `usePreloadedQuery/useQueryLoader`. L76 in particular feels wrong, but i was getting a blank screen when trying to load the page after logout.
- Obviously we don't want to print `...Loading` or `Loading ...`. It does appear on the screen long enough to notice, which I interpret to mean it adds a perceptible loading time.
- The refetch is intended to be used when someone takes an action that justifies this
- This could be useful for bringing in other global stuff (my bids, orders, notifications etc?) and we could adopt a deferred loading strategy for some things that don't need to be available on page load.